File: study/sieve_of_erathostenes.cc
#include <bits/stdc++.h>

using namespace std;

#define SIZE 60000000LL

int main() {
    bitset<SIZE+1> sieve;
    sieve[0] = 1;
    sieve[1] = 1;
    for(int i=2; i*i<=SIZE; i+=1) {
        if(sieve[i] == 1) continue;

        for(int j=i+i; j<=SIZE; j+=i)
            sieve[j] = 1;
    }

    for(int i=2; i<=SIZE; i++)
        if(sieve[i] == 0)
            cout << "prime: " << i << endl;
    return 0;
}

File: study/sieve_of_erathostenes.rb

SIZE = 1000000
sieve = Array.new(SIZE+1, true)

sieve[0] = false
sieve[1] = false

SIZE.times do |i|
  if sieve[i]
    k = i+i
    while k <= SIZE
      sieve[k] = false
      k += i
    end
  end
end

SIZE.times do |i|
  if sieve[i]
    puts "Prime: #{i}"
  end
end
